[SQUASHME]ASoC: SOF: fix return value for sof_link_load

ranj063 opened the pull request:
Return the correct value instead of 0.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>

plbossart commented on sound/soc/sof/topology.c:
```diff
@@ -1749,7 +1749,7 @@ static int sof_link_load(struct snd_soc_component *scomp, int index,
 		       sizeof(struct sof_ipc_dai_config));
 
-	return 0;
+	return ret;
```
wondering if we should also return in the default case line 1743/44, it makes no sense to provide an error message and then still call snd_sof_find_dai.

ranj063 commented:
@plbossart I've updated the commit to handle the error before setting dai config. I've also added an error if we cannot find a valid dai for the link.
